From 0a49ed4d323bbc93764b8a3e3b7e9e4f7fe72736 Mon Sep 17 00:00:00 2001
From: Nathan Muggli <nmuggli@google.com>
Date: Thu, 7 May 2026 11:25:20 -0600
Subject: [PATCH] disable boringssl in base::RandBytes and
 NonAllocatingRandomBitGenerator

Introduced in r1045239 (crrev.com/c/3880863).
Updated in r1321254 (crrev.com/c/5669584).
Updated in r1477790 (crrev.com/c/6665689).
Updated in r1624070 (crrev.com/c/7812144).
Updated in r1626560 (crrev.com/c/7828277).
Updated in r1632053 (crrev.com/c/7856143).
Updated in r1682187 (crrev.com/c/8265040).
Updated in r1718350 (crrev.com/c/8578019).

Also fold in long-term-2000 (disable boringssl used in
base::NonAllocatingRandomBitGenerator in base/rand_util.h) since both
remove <openssl/rand.h> from base/rand_util.

Change-Id: I465dc418929f7ed936d5a2c8be114ca8194acc7e
patch-name: long-term-0800-disable-boringssl-in-base-RandBytes.patch
---
 base/rand_util.h        | 17 -----------------
 base/rand_util_posix.cc | 10 ----------
 2 files changed, 27 deletions(-)

diff --git a/base/rand_util.h b/base/rand_util.h
index ff59e32c84..cb8f86657c 100644
--- a/base/rand_util.h
+++ b/base/rand_util.h
@@ -23,7 +23,6 @@
 #include "base/numerics/safe_conversions.h"
 #include "base/time/time.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/rand.h"
 
 namespace memory_simulator {
 class MemoryHolder;
@@ -186,22 +185,6 @@ class RandomBitGenerator {
   ~RandomBitGenerator() = default;
 };
 
-class NonAllocatingRandomBitGenerator {
- public:
-  using result_type = uint64_t;
-  static constexpr result_type min() { return 0; }
-  static constexpr result_type max() { return UINT64_MAX; }
-  result_type operator()() const {
-    uint64_t result;
-    RAND_get_system_entropy_for_custom_prng(reinterpret_cast<uint8_t*>(&result),
-                                            sizeof(result));
-    return result;
-  }
-
-  NonAllocatingRandomBitGenerator() = default;
-  ~NonAllocatingRandomBitGenerator() = default;
-};
-
 // Shuffles [first, last) randomly. Thread-safe.
 template <typename Itr>
 void RandomShuffle(Itr first, Itr last) {
diff --git a/base/rand_util_posix.cc b/base/rand_util_posix.cc
index 71bb35a2ff..697d6817c0 100644
--- a/base/rand_util_posix.cc
+++ b/base/rand_util_posix.cc
@@ -12,7 +12,6 @@
 #include "base/containers/span.h"
 #include "base/time/time.h"
 #include "build/build_config.h"
-#include "third_party/boringssl/src/include/openssl/rand.h"
 
 #if BUILDFLAG(IS_MAC)
 #include <sys/random.h>
@@ -30,10 +29,6 @@
 #include "base/system/sys_info.h"
 #endif  // BUILDFLAG(IS_MAC)
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-#include "third_party/lss/linux_syscall_support.h"
-#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-
 namespace base {
 
 namespace {
@@ -93,11 +88,6 @@ bool GetRandomSyscall(void* output, size_t output_length) {
         // BUILDFLAG(IS_ANDROID)
 
 void RandBytesInternal(span<uint8_t> output, bool avoid_allocation) {
-  if (!avoid_allocation) {
-    // BoringSSL's RAND_bytes always returns 1. Any error aborts the program.
-    (void)RAND_bytes(output.data(), output.size());
-    return;
-  }
 #if BUILDFLAG(IS_MAC)
   CHECK(getentropy(output.data(), output.size()) == 0);
 #elif BUILDFLAG(IS_APPLE)
-- 
2.56.0.385.gd3acb90ef8-goog

